### DISCUSSION TITLE
[BUG] Add-PnPEntraIDServicePrincipalAppRole no response

### DISCUSSION THREAD

**Discussion by eddison-shyi** [pnp/powershell#4356] [General]
### Reporting an Issue or Missing Feature
Get-PnPAzureADServicePrincipalAssignedAppRole & Add-PnPEntraIDServicePrincipalAppRole no response (Same with Azure CLI & Windows PowerShell 7), and after 30min it shows verification code expired error. (I've logged in successfully before calling these command)
![image](https://github.com/user-attachments/assets/8c0ceb33-8575-48a9-865a-a60312511735)

### Expected behavior 
Return the result immediately

### Actual behavior 
Idle until verification code expired

### Steps to reproduce behavior
```powershell
# Make sure ClientId app with proper permission for next command, e.g. AppRoleAssignment.ReadWrite.All and Application.Read.All
Connect-PnPOnline -Url ...  -DeviceLogin -ClientId ... -Tenant ...

# After login success
Add-PnPEntraIDServicePrincipalAppRole -Principal $AppID -AppRole "Sites.FullControl.All" -BuiltInType SharePoint
```
### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)
2.12.0 on powershell 7.4.5 (windows)

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows
- [ ] Linux
- [ ] MacOS
- [X] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
Seems like some kind of authentication issue with your EntraID app and not connected to the cmdlet itself.

You need to assign `AppRoleAssignment.ReadWrite.All` and ` Application.Read.All` to the app you are using in Connect-PnPOnline and then use this cmdlet.
